src/library/scala/collection/mutable/OrderAwareHashMap.scala

```diff
@@ -0,0 +1,800 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection
+package mutable
+
+import scala.annotation.meta.{getter, setter}
+import scala.annotation.tailrec
+import scala.collection.generic.DefaultSerializationProxy
+import scala.collection.mutable
+
+/** This class implements mutable sets using a hashtable with red-black trees in the buckets for good
```

Comment:
Mutable maps you mean?

Reply:
Nice catch. This started out as an `OrderAwareHashSet` but I quickly realized that it's getting so big to the point where I wouldn't want to maintain separate map and set implementation so I switched over to a map right away :-)